src/gl/mod.rs

```diff
@@ -1067,6 +1068,41 @@ impl Renderer {
                         .upload_raw(GenMipmaps::No, body)
                         .map_err(Error::Texture)?;
                 }
+                ViewOp::Flip(layer_id, src, dir) => {
+                    let resources = self.resources.lock();
+                    let (_, mut pixels) = resources.get_snapshot_rect(id, *layer_id, src).unwrap(); // TODO: Handle this nicely?
+                    let (w, h) = (src.width() as u32, src.height() as u32);
+                    let [paste_w, paste_h] = self.paste.size();
+
+                    if paste_w != w || paste_h != h {
+                        self.paste =
+                            Texture::new(&mut self.ctx, [w as u32, h as u32], 0, self::SAMPLER)
+                                .map_err(Error::Texture)?;
+                    }
+
+                    match dir {
+                        Flip::Vertical => {
+                            let len = pixels.len();
+
+                            let (front, back) = pixels.split_at_mut(len / 2);
+                            for (front_row, back_row) in front
+                                .chunks_exact_mut(w as usize)
+                                .zip(back.rchunks_exact_mut(w as usize))
+                            {
+                                front_row.swap_with_slice(back_row);
+                            }
```

Comment:
Very elegant solution :+1:

Reply:
Thank you, it's the result of brainstorming with guys on Discord, haha